hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -633,6 +663,10 @@ def validate_health_check_suppressions(suppressions):
 
 def _validate_deadline(x):
     if x is None or isinstance(x, (int, float)):
+        if x is not None and x < 0:
+            raise InvalidArgument(
+                "deadline=%r must be positive; pass deadline=None to disable" % (x,)
```

Comment:
I would call `>= 0` non-negative rather than positive but also I wonder if we should make `deadline=0` an error. Arguably backwards compatible in that every test with `deadline=0` must fail.

Reply:
I'm inclined to allow it, since you'll get the same DeadlineExceeded error from `+eps`.  This is really just to catch the typical "-1 disables timeouts" early.

Reply:
I don't feel especially strongly on the subject, so feel free to leave it as is if you still disagree, but I feel like it's quite plausible that someone would expect that setting a deadline to 0 would disable the deadline.

A thing that's just occurred to me on top of this is that we probably _do_ need `try_convert` for backwards compatibility in there, it is definitely plausible that people would pass `deadline=False`, and post-`try_convert` this means the same as `deadline=0.0`.

"I hate everything, let the world burn" is a valid response to this design consideration. 😢

Reply:
😭

OK, `deadline=0` will be an error.  `deadline=True` will be deprecated; and `deadline=False` will be deprecated **and** an error.  I know at least some of our users appreciate this though, so...